### PR TITLE
PIMS-1343,PIMS-1344,PIMS-1383

### PR DESCRIPTION
**Pull request by devinleighsmith**
Only display a single view/update button on parcel popup detail depending on user agency ownership.

### CODE REVIEW

File: frontend/src/actionCreators/parcelsActionCreator.spec.tsx [hunk 1]
```diff
@@ -27,9 +27,9 @@ describe('fetchParcels action creator', () => {
     const mockResponse = { data: { success: true } };
     mockAxios.onGet(url).reply(200, mockResponse);
     return fetchParcels(null)(dispatch).then(() => {
-      expect(requestSpy).toHaveBeenCalledTimes(0);
-      expect(successSpy).toHaveBeenCalledTimes(0);
-      expect(dispatch).toHaveBeenCalledTimes(0);
+      expect(requestSpy).toHaveBeenCalledTimes(1);
```
Comment (Fosol) [inline frontend/src/actionCreators/parcelsActionCreator.spec.tsx:30]: I'm not sure this is correct.  I had to change them to 0 when `null` is passed to `fetchParcels(...)` for the test to pass.  But I could be doing something wrong.

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx [hunk 2]
```diff
@@ -44,13 +46,12 @@ const mockDetails: IParcelDetail = {
     isSensitive: false,
     evaluations: [
       {
-        assessedValue: 1000000,
-        estimatedValue: 0,
-        fiscalYear: 2019,
-        netBookValue: 0,
-        appraisedValue: 0,
+        date: '2019',
+        key: '',
```
Comment (Fosol) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx:50]: Did you want to test an empty string?  I thought I created enums that you could use.

File: frontend/src/pages/SubmitProperty.test.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    latitude: 48,
+    longitude: 123,
+    propertyStatus: 'active',
+    classification: 'Core Operational',
+    description: 'test',
+    isSensitive: false,
+    evaluations: [
+      {
+        date: '2019',
+        key: '',
```
Comment (Fosol) [inline frontend/src/pages/SubmitProperty.test.tsx:49]: Did you want to test an empty string?  I thought I created enums that you could use.

File: frontend/src/actionCreators/parcelsActionCreator.spec.tsx (hunk 1 above)
Comment (devinleighsmith) [reply frontend/src/actionCreators/parcelsActionCreator.spec.tsx:30]: The two scenarios that we support are 
1) null boundaries
or
2) boundaries where the area is bigger then a point
So this should have a request, success, etc.

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx (hunk 2 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/MapProperties.test.tsx:50]: This is just to make this test compile, I'll write actual tests for this change when I start working on the parcelDetailForm updates.

File: frontend/src/components/maps/ParcelPopupView.tsx [hunk 4]
```diff
@@ -57,14 +59,13 @@ export const ParcelPopupView = (props: IParcelDetailProps | null) => {
...
-                Update
-              </Link>
-            )}
+            {parcelDetail?.id &&
+              !props?.disabled &&
+              (!keycloak.hasAgency(parcelDetail?.agencyId) ? (
+                <Link to={`/submitProperty/${parcelDetail?.id}?disabled=true`}>View</Link>
+              ) : (
+                <Link to={`/submitProperty/${parcelDetail?.id}`}>Update</Link>
+              ))}
```
Comment (asanchezr) [inline frontend/src/components/maps/ParcelPopupView.tsx:68]: The logic here seems a bit hard to follow to me. 
What would you think of maybe having two booleans `canView`, `canUpdate` and then JSX looks a bit cleaner IMO?

```jsx
{canView && (<Link ... />)}
{canUpdate && (<Link ... />)}
```

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 5]
```diff
@@ -72,6 +72,11 @@ const Map: React.FC<MapProps> = ({
   const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
   const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);
 
+  //do not jump to map coordinates if we have an existing map but no parcel details.
+  if (mapRef.current && !selectedProperty?.parcelDetail) {
+    lat = (mapRef.current.props.center as Array<number>)[0];
+    lng = (mapRef.current.props.center as Array<number>)[1];
+  }
```
Comment (asanchezr) [inline frontend/src/components/maps/leaflet/Map.tsx:79]: I'm okay if this PR gets merged as it has already been approved but the above doesn't seem to fix the jumping around on the map. I will post a video shortly.

**Comment by asanchezr** [comment]
@devinleighsmith I still see the map jumping around bug. No need to block this PR, just need to be handled properly in another PR

video

File: frontend/src/components/maps/leaflet/Map.tsx (hunk 5 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/Map.tsx:79]: We talked about this in rocketchat, this PR fixes two issues:
1) The map moving when the popup is closed.
2) The map only opening popups once.

This does not fix the issue assigned to Goeme PIMS-1549 (and isn't intended to).
